homeassistant/components/mqtt/number.py

```diff
@@ -160,22 +167,28 @@ class MqttNumber(MqttEntity, RestoreNumber):
     _entity_id_format = number.ENTITY_ID_FORMAT
     _attributes_extra_blocked = MQTT_NUMBER_ATTRIBUTES_BLOCKED
 
-    def __init__(self, hass, config, config_entry, discovery_data):
-        """Initialize the MQTT Number."""
-        self._config = config
-        self._optimistic = False
-        self._sub_state = None
+    _optimistic: bool
+    _templates: dict[str, Callable[..., Any]]
```

Comment:
Maybe this should be split into two?
```suggestion
    _command_templates: dict[str, Callable[[PublishPayloadType], PublishPayloadType]]
    _value_templates: dict[str, Callable[[ReceivePayloadType], ReceivePayloadType]]
```